rpc/handlers.go

```diff
@@ -110,17 +111,41 @@ func (h *Handler) BlockWithTxHashes(id BlockID) (*BlockWithTxHashes, *jsonrpc.Er
 		txnHashes[index] = txn.Hash()
 	}
 
+	l1H, err := h.l1Head()
+	if err != nil {
+		return nil, ErrInternal
+	}
+
 	status := StatusAcceptedL2
-	if id.Pending {
+	if block.Hash != nil && isL1Verified(block.Number, block.Hash, l1H) {
+		status = StatusAcceptedL1
+	} else if id.Pending {
 		status = StatusPending
 	}
+
 	return &BlockWithTxHashes{
 		Status:      status,
 		BlockHeader: adaptBlockHeader(block.Header),
 		TxnHashes:   txnHashes,
 	}, nil
 }
 
+func (h *Handler) l1Head() (*core.L1Head, error) {
+	l1Head, err := h.bcReader.L1Head()
+	if err != nil && !errors.Is(err, db.ErrKeyNotFound) {
+		return nil, err
+	}
+	// nil is returned if l1 head doesn't exist
+	return l1Head, nil
+}
+
+func isL1Verified(n uint64, h *felt.Felt, l1 *core.L1Head) bool {
+	if l1 != nil && l1.BlockHash.Equal(h) && l1.BlockNumber >= n {
+		return true
+	}
```

Comment:
I am not sure which scenario you are referring to

Reply:
Since it is a private func is it not reasonable to make the assumption the `h` and `n` will be of the same block?

I initially passed the block but Transaction Receipt by Hash doesn't have access to the full block. We would need to make another db call to get it. So that is why i passed the block hash and number separately.

Reply:
that is not what I am trying to say. If the block you are trying to check its L1 verification status (let's call it `target` block) is not the L1 head itself, this function will always return false. You are checking that the `target` block and the l1 head has the same hash

Reply:
I understand, we can only compare block numbers and not hashes. I am not sure why I did that, I initially only checked the block number.

Reply:
fixed